package.json

```diff
@@ -135,7 +135,49 @@
         "language": "solidity",
         "path": "./snippets/solidity.json"
       }
-    ]
+    ],
+    "taskDefinitions": [
+      {
+        "type": "hardhat",
+        "required": [
+          "task"
+        ],
+        "properties": {
+          "task": {
+            "type": "string",
+            "description": "%taskdef.task%"
+          }
+        },
+        "when": "shellExecutionSupported"
+      }
+    ],
+    "commands": [
+      {
+        "command": "hardhat.solidity.compile",
+        "title": "Hardhat: Compile project"
+      },
+      {
+        "command": "hardhat.solidity.clean",
+        "title": "Hardhat: Clear cache and artifacts "
+      },
+      {
+        "command": "hardhat.solidity.flattenCurrentFile",
+        "title": "Hardhat: Flatten this file and their dependencies",
+        "enablement": "editorLangId == 'solidity'"
+      }
+    ],
+    "menus": {
+      "editor/context": [
+        {
+          "when": "editorLangId == 'solidity'",
+          "command": "hardhat.solidity.compile"
```

Comment:
I don't think we should offer a project compile here. I think a file compile does make sense, but I don't know if hardhat offers this.

Reply:
No problem, I'll remove it. You are right, hardhat doesn't support individual contract compilations out of the box